llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_vop2.txt

```diff
@@ -446,49 +446,78 @@
 
 0x7e,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00
-# GFX12: v_fmaak_f16 v5, exec_lo, v2, 0xfe0b     ; encoding: [0x7e,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-REAL16: v_fmaak_f16 v5.l, exec_lo, v2.l, 0xfe0b ; encoding: [0x7e,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v5, exec_lo, v2, 0xfe0b     ; encoding: [0x7e,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
 
 0x7f,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00
-# GFX12: v_fmaak_f16 v5, exec_hi, v2, 0xfe0b     ; encoding: [0x7f,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-REAL16: v_fmaak_f16 v5.l, exec_hi, v2.l, 0xfe0b ; encoding: [0x7f,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v5, exec_hi, v2, 0xfe0b     ; encoding: [0x7f,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
 
 0x7c,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00
-# GFX12: v_fmaak_f16 v5, null, v2, 0xfe0b        ; encoding: [0x7c,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-REAL16: v_fmaak_f16 v5.l, null, v2.l, 0xfe0b    ; encoding: [0x7c,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v5, null, v2, 0xfe0b        ; encoding: [0x7c,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
 
 0xc1,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00
-# GFX12: v_fmaak_f16 v5, -1, v2, 0xfe0b          ; encoding: [0xc1,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-REAL16: v_fmaak_f16 v5.l, -1, v2.l, 0xfe0b      ; encoding: [0xc1,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v5, -1, v2, 0xfe0b          ; encoding: [0xc1,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
 
 0xf0,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00
-# GFX12: v_fmaak_f16 v5, 0.5, v2, 0xfe0b         ; encoding: [0xf0,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-REAL16: v_fmaak_f16 v5.l, 0.5, v2.l, 0xfe0b     ; encoding: [0xf0,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v5, 0.5, v2, 0xfe0b         ; encoding: [0xf0,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
 
 0xfd,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00
-# GFX12: v_fmaak_f16 v5, src_scc, v2, 0xfe0b     ; encoding: [0xfd,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-REAL16: v_fmaak_f16 v5.l, src_scc, v2.l, 0xfe0b ; encoding: [0xfd,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v5, src_scc, v2, 0xfe0b     ; encoding: [0xfd,0x04,0x0a,0x70,0x0b,0xfe,0x00,0x00]
 
 0xff,0xfe,0xfe,0x70,0x0b,0xfe,0x00,0x00
-# GFX12: v_fmaak_f16 v127, 0xfe0b, v127, 0xfe0b  ; encoding: [0xff,0xfe,0xfe,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-REAL16: v_fmaak_f16 v127.l, 0xfe0b, v127.l, 0xfe0b ; encoding: [0xff,0xfe,0xfe,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v127, 0xfe0b, v127, 0xfe0b  ; encoding: [0xff,0xfe,0xfe,0x70,0x0b,0xfe,0x00,0x00]
+
+0x81,0x05,0x0a,0x70,0x0b,0xfe,0x00,0x00
+# GFX12-REAL16: v_fmaak_f16 v5.l, v1.h, v2.l, 0xfe0b    ; encoding: [0x81,0x05,0x0a,0x70,0x0b,0xfe,0x00,0x00]
+# GFX12-FAKE16: v_fmaak_f16 v5, v129/*Invalid register, operand has 'VS_32_Lo128' register class*/, v2, 0xfe0b ; encoding: [0x81,0x05,0x0a,0x70,0x0b,0xfe,0x00,0x00]
```

Comment:
I think we are deleting the fake16 check lines for bytes encoding .h registers? So there are are few to delete in every disassembly file.

Reply:
I am currently updating the test with the update_mc script and the result depends on the llvm-mc output in stdout. I think there are three output now:
1. dasm gives nothing in fake16 when .h is used
2. dasm gives an error message in fake16 when .h in use, which is the case here
3. dasm ignore the .h and generate all .l instructions

I haven't check in details to see why, but my impression is that it related with the number of bytes of these instructions. 

Do you think it's ok we leave it as it? as should we further remove the fake16 check lines for .h register?

Reply:
This sounds to me like the behavior of putting bytes that encode .h registers into the fake16 disassembler is undefined. So we should not have tests of undefined behavior. So remove the Fake16 check lines for those tests. @kosarev what do you think?

Reply:
Might still make sense to make sure we don't do nasty things like crashing or something on .h operands in Fake16 instructions?

Reply:
I think the output is confusing and not useful. For example the following. The disassembler did not work on the full 8 bytes and spit out the last 4 bytes disassembled. 

```
0xfa,0x04,0x0b,0x6d,0x81,0x60,0x01,0x13
# GFX12-REAL16: v_fmac_f16_dpp v5.h, v1.h, v2.h row_xmask:0 row_mask:0x1 bank_mask:0x3 ; encoding: [0xfa,0x04,0x0b,0x6d,0x81,0x60,0x01,0x13]
# GFX12-FAKE16: v_mul_i32_i24_e32 v128, 1, v176         ; encoding: [0x81,0x60,0x01,0x13]
```
Crashing is perhaps the only behavior worth testing, and that could be done with --not-crash in the runline.

Reply:
I agree with Joe that these fake16 tests are pointless and I was just thinking it might worth to maintain it at this time. But if you both think it's too messy here then I am ok with it.

 If I will remove this I will recommend to do it in a separate patch and remove all of them in all test files.

Reply:
How the script is going to know which check lines should be dropped?

Reply:
I think this might need to be clean up manually, or the dasm will need to be fixed to error out

Reply:
I think I would rather prefer having the extra check lines than removing them manually on every update using the script.

Reply:
Do you think it's a likely scenario that someone makes a change that affects a meaningless fake16 test line and doesn't know how to remediate that? Perhaps they will just run the update script and be happy. 

Alternatively, can the update script can be enhanced to support automatically removing these cases? Some logic like this: If REAL16 and FAKE16 lines differ after removing all instances of '.l' and '.h' from the REAL16 line, delete the FAKE16 line check line (because the test is for REAL16 bytes)

Reply:
> Do you think it's a likely scenario that someone makes a change that affects a meaningless fake16 test line and doesn't know how to remediate that? Perhaps they will just run the update script and be happy.

I just think  the burden of manual editing is a larger practical problem than having the less-useful-than-usual check lines. Looks like having them wasn't an issue so far and we know they are temporary anyway.

Reply:
Ok, I'm convinced. Landing this with the less useful fake16 checks is ok with me.